db/queries.py

```diff
@@ -3,44 +3,76 @@
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return [
+        message for message in Message.objects.filter(text__icontains=word)
+    ]
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return [
+        chat for chat in Chat.objects.filter(title__startswith="Untitled")
+    ]
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return [user for user in Message.objects.filter(
+        sent__year=2015
+    ).values_list("user__first_name", "user__last_name")]
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return Chat.objects.filter(message__sent__year__gt=2020)
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return Message.objects.filter(text__icontains=F("user__first_name"))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return User.objects.filter(
+        Q(message__text__istartswith="a") | Q(message__text__istartswith="m")
+    )
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return Message.objects.filter(
+        Q(user__username__startswith="admin") | Q(is_delivered=1)
+    )
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return sum(
```

Comment:
You can greatly simplify this function. You're going to need need just one query and the `.count()` method.

Reply:
Thank you, its really great method.